src/reqstool/commands/report/report.py

```diff
@@ -122,43 +90,69 @@ def __generate_asciidoc_information(self, reqs, statistics: StatisticsContainer)
         Returns:
             str : All data rendered as AsciiDoc
         """
-        statistics_table = self.__render(statistics._total_statistics, self.templates[TemplateNames.TOTAL_STATISTICS])
-
-        result = "== REQUIREMENT DOCUMENTATION\n"
-        initial = "=== INITIAL REQUIREMENTS\n"
-        imported = "=== IMPORTED REQUIREMENTS\n"
-        # we should start to generate requirements defined in inital source
-        for req_template in reqs["templates"]:
-            if reqs["initial_model"] == req_template["urn"]:
-                initial += self.__extract_template_data(req_template=req_template)
-            else:
-                imported += self.__extract_template_data(req_template=req_template)
-        return result + statistics_table + initial + imported
+        statistics_table = Jinja2Utils.render(
+            data=statistics._total_statistics, template=self.jinja2_templates[Jinja2Templates.TOTAL_STATISTICS]
+        )
+
+        grouped_requirements: Dict[str, List[UrnId]] = GroupByOrganizor(
+            cid=cid, group_by=self.group_by, sort_by=self.sort_by
+        ).grouped_requirements
+
+        # category, List(asciidoc for each req)
+        template_data: Dict[str, List[str]] = defaultdict(list)
```

Comment:
Sonar complains about this line as it is never used. 

Perhaps it is better remove this line and to combine the type help with the declaration of the dict below instead? 


        template_data: Dict[str, List[str]] = {
            category: [self.__extract_template_data(req_template=aggregated_data[urn_id]) for urn_id in urn_ids]
            for category, urn_ids in grouped_requirements.items()
        }

Reply:
Fixed already. Pushing soon.